tests/test_client.py

```diff
@@ -445,6 +473,198 @@ def test_client_site_selection_by_name(config_toml: Path) -> None:
+
+
+@pytest.mark.usefixtures("disable_dot_env")
+@pytest.mark.skipif(not is_conda_installed(), reason="Conda module not available")
+def test_client_kwargs_supremecy(config_toml: Path) -> None:
+    from tests.conda_token.test_condarc import make_temp_condarc
+
+    config_toml.write_text(
+        dedent(
+            """\
+                [sites.local]
+                domain = "localhost"
+                auth_domain_override = "auth-local"
+                ssl_verify = false
+                api_key = "foo"
+
+
+                [plugin.auth.proxy_servers]
+                http = "toml"
+                https = "toml"
+                """
+        )
+    )
+
+    original_condarc = dedent(
+        """\
+        ssl_verify: truststore
+        proxy_servers:
+          http: condarc
+          https: condarc
+
+        default_channels:
+          - https://repo.anaconda.com/pkgs/main
+        channels:
+          - defaults
+          - conda-forge
+
+        channel_alias: https://conda.anaconda.org/
+        """
+    )
+
+    with make_temp_condarc(original_condarc):
+        client = BaseClient(proxy_servers={"http": "kwargy", "https": "kwargy"})
+        assert client.config.ssl_verify
+        assert client.proxies["http"] == "kwargy"
+        assert client.proxies["https"] == "kwargy"
+
+
+@pytest.mark.usefixtures("disable_dot_env")
+@pytest.mark.skipif(not is_conda_installed(), reason="Conda module not available")
+def test_client_ssl_context(config_toml: Path) -> None:
+    from tests.conda_token.test_condarc import make_temp_condarc
+
+    original_condarc = dedent(
+        """\
+        ssl_verify: truststore
+        proxy_servers:
+          http: condarc
+          https: condarc
+
+        default_channels:
+          - https://repo.anaconda.com/pkgs/main
+        channels:
+          - defaults
+          - conda-forge
+
+        channel_alias: https://conda.anaconda.org/
+        """
+    )
+
+    with make_temp_condarc(original_condarc):
+        import ssl
+
+        import truststore  # type: ignore
+
+        ssl_context = truststore.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+
+        client = BaseClient()
+        assert client.config.ssl_verify
+        # TODO change this to some type of equality operator
+        assert (
+            client.adapters["http://"]._ssl_context._ctx.protocol
+            == ssl_context._ctx.protocol
+        )
+        assert (
+            client.adapters["https://"]._ssl_context._ctx.protocol
+            == ssl_context._ctx.protocol
+        )
+
+
+@pytest.mark.usefixtures("disable_dot_env")
+@pytest.mark.skipif(not is_conda_installed(), reason="Conda module not available")
+def test_client_condarc_certs(config_toml: Path) -> None:
+    from tests.conda_token.test_condarc import make_temp_condarc
+
+    original_condarc = dedent(
+        """\
+        ssl_verify: truststore
+        client_cert: client_cert.pem
+        client_cert_key: client_cert_key
```

Comment:
sorry if its maybe too much, but would we also want to support `client_cert` and `client_cert_key` in the AnacondaAuthSite pydantic model with all the same overrides?